### PR TITLE
New Models, Hyperparameter Search

### PR DESCRIPTION
**Pull request by liaopeiyuan**
- New models inclusing MLP (from @ruochen99), [TabNet](https://arxiv.org/abs/1908.07442) and [LightGBM](https://papers.nips.cc/paper/2017/file/6449f44a102fde848669bdd9eb6b76fa-Paper.pdf)
- XGB Hyperparameter Search using NNI
- LightGBM Hyperparameter Search using Bayesian optimization

Performance on default parameter: LightGBM vs XGBoost:

Model: lgbm
RMSE: 0.0791
R^2: 0.7553
pairwise comparision accuracy: 0.8526
mape: 10263628506.3581
average peak score@1: 0.8377
average peak score@5: 0.9096

Model: xgb
RMSE: 0.0799
R^2: 0.7507
pairwise comparision accuracy: 0.8503
mape: 10027810410.0418
average peak score@1: 0.8240
average peak score@5: 0.9034

TabNet vs MLP (on a reduced training set of 1000 graphs due to memory limits):

Model: tab
RMSE: 7.7406
R^2: -2577.6602
pairwise comparision accuracy: 0.8830
mape: 1508452198603.7810
average peak score@1: 0.8839
average peak score@5: 0.9469

Model: mlp
RMSE: 7.8012
R^2: -2617.3268
pairwise comparision accuracy: 0.8703
mape: 1438344501262.7888
average peak score@1: 0.8636
average peak score@5: 0.9346

end2end search on ResNet-50:

XGBoost:
Estimated total latency: 18.435 ms      Trials: 102     Used time : 199 s       Next ID: -1
Mean inference time (std dev): 15.20 ms (0.01 ms)

LightGBM:
Estimated total latency: 17.585 ms      Trials: 102     Used time : 194 s       Next ID: -1
Mean inference time (std dev): 15.16 ms (0.05 ms)

TabNet:
Estimated total latency: 20.950 ms      Trials: 102     Used time : 306 s       Next ID: -1
Mean inference time (std dev): 18.44 ms (0.07 ms)

MLP:

### CODE REVIEW

**Comment by liaopeiyuan** [comment]
This PR is ready for review & merge.
